sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlockBlobClient.java

```diff
@@ -220,6 +224,35 @@ public BlobOutputStream getBlobOutputStream(BlockBlobOutputStreamOptions options
         return BlobOutputStream.blockBlobOutputStream(blobClient, options, null);
     }
 
+    /**
+     * Opens a seekable byte channel in write-only mode to upload the blob.
+     *
+     * @param options {@link BlobSeekableByteChannelReadOptions}
+     * @return A <code>SeekableByteChannel</code> object that represents the channel to use for writing to the blob.
+     * @throws BlobStorageException If a storage service error occurred.
+     */
+    public SeekableByteChannel openSeekableByteChannelWrite(BlockBlobSeekableByteChannelWriteOptions options) {
+        Objects.requireNonNull(options);
+
+        // Behavior can support more modes but this client does not currently support them
+        StorageSeekableByteChannelBlockBlobWriteBehavior.WriteMode internalMode;
+        long startingPosition = 0L;
+        if (options.getWriteMode() == BlockBlobSeekableByteChannelWriteOptions.WriteMode.OVERWRITE) {
+            internalMode = StorageSeekableByteChannelBlockBlobWriteBehavior.WriteMode.OVERWRITE;
+        } else {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                "Unsupported value for `options.getWriteMode()`."));
+        }
+
+        return new StorageSeekableByteChannel(
+            options.getChunkSize() != null
+                ? options.getChunkSize().intValue()
+                : BlobAsyncClient.BLOB_DEFAULT_UPLOAD_BLOCK_SIZE,
```

Comment:
This will be a future enhancement, but long term we should support writes larger than 2 GB, likely through managing multiple `ByteBuffer`s internally. I don't believe this is something we should hold the feature back on before GAing it.

Reply:
Agreed. API uses a Long for this purpose.